src/cephadm/cephadm

```diff
@@ -2292,10 +2292,34 @@ def command_version():
 @infer_image
 def command_pull():
     # type: () -> int
-    logger.info('Pulling latest %s...' % args.image)
-    call_throws([container_path, 'pull', args.image])
+
+    _pull_image(args.image)
     return command_inspect_image()
 
+
+def _pull_image(image) -> None:
+    logger.info('Pulling container image %s...' % image)
+
+    ignorelist = [
+        "error creating read-write layer with ID",
+        "net/http: TLS handshake timeout",
+    ]
+
+    cmd = [container_path, 'pull', image]
+
+    retry = args.retry
+
+    for count in range(retry):
+        out, err, ret = call(cmd)
+        if not ret:
+            return
+
+        if not any(pattern in err for pattern in ignorelist):
+            raise RuntimeError('Failed command: %s' % ' '.join(cmd))
+
+        time.sleep(min(count, 3))
```

Comment:
so the wait between retries is always something between 1 and 3 seconds?

I'm thinking that might be too short...

Reply:
`DEFAULT_RETRIES=10`. which means we're waiting close to 30 seconds in total. waiting any longer would be really bad UX

Reply:
OK. Since "podman pull" might run for quite some seconds before it fails transiently, any failed attempts will work as a form of "sleep".

Reply:
that was my thought. waiting any longer would be bad imo.

Reply:
> `DEFAULT_RETRIES=10`. which means we're waiting close to 30 seconds in total. waiting any longer would be really bad UX

or even much longer since we haven't added any handling for `args.timeout` ..